tests/strategies/test_deepspeed_strategy.py

```diff
@@ -1269,13 +1269,19 @@ def test_deepspeed_with_meta_device(tmpdir):
 def test_deepspeed_multi_save_same_filepath(tmpdir):
     """Test that verifies that deepspeed saves only latest checkpoint in the specified path and deletes the old
     sharded checkpoints."""
-    model = BoringModel()
+
+    class CustomModel(BoringModel):
+        def training_step(self, *args, **kwargs):
+            self.log("grank", self.global_rank)
+            return super().training_step(*args, **kwargs)
+
+    model = CustomModel()
     trainer = Trainer(
         default_root_dir=tmpdir,
         strategy="deepspeed",
         accelerator="gpu",
         devices=2,
-        callbacks=[ModelCheckpoint(save_top_k=1, save_last=True)],
+        callbacks=[ModelCheckpoint(filename="{epoch}_{step}_{grank}", save_top_k=1, save_last=True)],
```

Comment:
Is epoch and step relevant to this test?

```suggestion
        callbacks=[ModelCheckpoint(filename="{grank}", save_top_k=1, save_last=True)],
```

Reply:
And `save_last`?

```suggestion
        callbacks=[ModelCheckpoint(filename="{epoch}_{step}_{grank}", save_top_k=1)],
```

Reply:
`save_last` is not required now.

I'd keep `step` to ensure the new checkpoint path is not the same as the old one and the old one is deleted too.